target/linux/ipq40xx/image/Makefile

```diff
@@ -64,6 +64,21 @@ define Device/avm_fritzbox-4040
 endef
 TARGET_DEVICES += avm_fritzbox-4040
 
+define Device/compex_wpj428
+	$(call Device/FitImage)
+	DEVICE_DTS := qcom-ipq4028-wpj428
+	DEVICE_DTS_CONFIG := config@4
+	BLOCKSIZE := 64k
+	SUPPORTED_DEVICES := compex,wpj428
+	DEVICE_TITLE := Compex WPJ428
+	IMAGE_SIZE := 31232k
+	KERNEL_SIZE := 4096k
```

Comment:
Where does the KERNEL_SIZE limit comes from?

Reply:
From some older version which had otherwise problems to split the rootfs correctly. Can be removed now together with the pad-to.

Reply:
Just remembered that it was not only for the padding. But the bootipq command will load the HLOS (kernel) to  CONFIG_SYS_LOAD_ADDR (0x84000000) before it tries to actually run the bootm. And the information about the position/size is taken from smem (MIBIB). And the commands used to load the kernel on this system are:

    (IPQ40xx) # smem
    flash_type:             0x6
    flash_index:            0x0
    flash_chip_select:      0x0
    flash_block_size:       0x10000
    flash_density:          0x2000000
    partition table offset  0x0
    No.: Name             Attributes            Start             Size
      0: 0:SBL1           0x0000ffff              0x0          0x40000
      1: 0:MIBIB          0x002040ff          0x40000          0x20000
      2: 0:QSEE           0x0000ffff          0x60000          0x60000
      3: 0:CDT            0x0000ffff          0xc0000          0x10000
      4: 0:DDRPARAMS      0x0000ffff          0xd0000          0x10000
      5: 0:APPSBLENV      0x0000ffff          0xe0000          0x10000
      6: 0:APPSBL         0x0000ffff          0xf0000          0x80000
      7: 0:ART            0x0000ffff         0x170000          0x10000
      8: 0:HLOS           0x0000ffff         0x180000         0x400000
      9: rootfs           0x0000ffff         0x580000        0x1600000

    nand device 2
    sf probe && sf read 0x%x 0x%x 0x%x && 
    bootm 0x%x

This translates to something like:

    nand device 2
    sf probe && sf read 0x84000000 0x180000 0x400000 && 
    bootm 0x84000000

At least this is the code from the CA_NETWORKING_2016_SPF_3.0 CSU2-r00030.1 u-boot sources (arch/arm/cpu/armv7/qca/cmd_bootqca.c)